deflate.c

```diff
@@ -1470,17 +1467,15 @@ static block_state deflate_stored(deflate_state *s, int flush) {
                 memcpy(s->window, s->window + s->w_size, s->strstart);
                 if (s->matches < 2)
                     s->matches++;   /* add a pending slide_hash() */
-                if (s->insert > s->strstart)
-                    s->insert = s->strstart;
+                s->insert = MIN(s->insert, s->strstart);
             }
             memcpy(s->window + s->strstart, s->strm->next_in - used, used);
             s->strstart += used;
             s->insert += MIN(used, s->w_size - s->insert);
         }
         s->block_start = (int)s->strstart;
     }
-    if (s->high_water < s->strstart)
-        s->high_water = s->strstart;
+    s->high_water = MIN(s->high_water, s->strstart);
```

Comment:
I think this should be MAX().

Reply:
Good catch I appreciate it!

Reply:
You know... I go through most of pull requests if I just understand what the code does... There is still a lot of code I don't understand enough to be able to review.